### ISSUE TITLE
curly braces linter and comments

### ISSUE THREAD

**Issue by wamserma** [jimhester/lintr#188]
The curly braces linter wants a newline after an opening curly brace, not accepting comments there.

Example
```r
if ("P" != "NP") { # what most people expect
    print("Cryptomania is possible")
} else { # what would make a lot of people sad
    print("we live in Algorithmica")
}
```

lintr output:
```
R\test.R:7:22: style: Opening curly braces should never go on their own line and should always be followed by a new line.
    if ("P" != "NP") { # what most people expect
                     ^
R\test.R:9:12: style: Opening curly braces should never go on their own line and should always be followed by a new line.
    } else { # what would make a lot of people sad
```

**Comment by fangly** [comment]
Another case where this linter fails, in my opinion, is with standalone code blocks (without a leading if/for/while/repeat):
```
withr::with_options(
     list(verbose=TRUE),
     {
         print("hello")
         print("world")
     }
)
```

**Comment by prosoitos** [comment]
I second @wamserma (almost posted a duplicate to this actually :stuck_out_tongue: )
